src/scripts/main.js

```diff
@@ -1,3 +1,292 @@
+  required
+  value=""
+>
+</label>
+
+<label>
+  Salary:
+  <input 
+  name="salary" 
+  type="number" 
+  data-qa="salary"
+  required
+  value=""
+  >
+</label>
+
+<button type="submit">Save to table</button>
+</form>
+`);
+
+const buttonForm = document.querySelector('button');
+
+function getSalary(el) {
+  return el.replace(/\D/g, '');
+}
+
+let counterClickName = 0;
+let counterClickPosition = 0;
+let counterClickOffice = 0;
+let counterClickAge = 0;
+let counterClickSalary = 0;
+
+function resetCount() {
+  counterClickName = 0;
+  counterClickPosition = 0;
+  counterClickOffice = 0;
+  counterClickAge = 0;
+  counterClickSalary = 0;
+}
+
+document.querySelector('thead').addEventListener('click', (e) => {
+  const whereСlicked = e.target;
+  const tableBodyActive = document.querySelector('tbody');
+  const personsNow = [...tableBodyActive.querySelectorAll('tr')];
+  let personsRender = [];
+
+  function sortedAlphabet(person) {
+    return person.sort((a, b) => a.children[whereСlicked.cellIndex]
+      .innerText.localeCompare(b.children[whereСlicked.cellIndex].innerText));
+  }
+
+  switch (whereСlicked.innerText) {
+    case 'Name':
+      if (counterClickName > 0) {
+        personsRender = personsNow.reverse();
+        break;
+      }
+
+      resetCount();
+      counterClickName++;
+      personsRender = sortedAlphabet(personsNow);
+      break;
+
+    case 'Position':
+      if (counterClickPosition > 0) {
+        personsRender = personsNow.reverse();
+        break;
+      }
+
+      resetCount();
+      counterClickPosition++;
+      personsRender = sortedAlphabet(personsNow);
+      break;
+
+    case 'Office':
+      if (counterClickOffice > 0) {
+        personsRender = personsNow.reverse();
+        break;
+      }
+
+      resetCount();
+      counterClickOffice++;
+      personsRender = sortedAlphabet(personsNow);
+      break;
+
+    case 'Age':
+      if (counterClickAge > 0) {
+        personsRender = personsNow.reverse();
+        break;
+      }
+
+      resetCount();
+      counterClickAge++;
+
+      personsRender = personsNow.sort((a, b) => {
+        return +a.children[whereСlicked.cellIndex].innerText
+        - +b.children[whereСlicked.cellIndex].innerText;
+      });
+      break;
+
+    case 'Salary':
+      if (counterClickSalary > 0) {
+        personsRender = personsNow.reverse();
+        break;
+      }
+
+      resetCount();
+      counterClickSalary++;
+
+      personsRender = personsNow.sort((a, b) => getSalary(a
+        .children[whereСlicked.cellIndex].innerText)
+        - getSalary(b.children[whereСlicked.cellIndex].innerText));
+  }
+
+  tableBody.append(...personsRender);
+});
+
+tableBody.addEventListener('click', (e) => {
+  [...document.querySelectorAll('.active')]
+    .forEach(el => el.classList.remove('active'));
+
+  e.target.closest('tr').classList.add('active');
+});
+
+buttonForm.onclick = (e) => {
+  e.preventDefault();
+
+  const labels = [...document.querySelectorAll('label')];
```

Comment:
it's better to use `new FormData(form)` where form is your employees form HTML element